examples/mensuration_and_polling/poll_for_tasks.py

```diff
@@ -0,0 +1,120 @@
+import time
+import json
+from datetime import datetime, timedelta, timezone
+import rasterio
+from shapely.geometry import Polygon
+from label_studio_sdk.client import LabelStudio
+from label_studio_sdk import Client, Project
+from label_studio_sdk.data_manager import Filters, Column, Operator, Type
+
+
+def poll_for_completed_tasks_new(
+    ls: LabelStudio, project_id: int, freq_sec: int
+) -> list:
+    while True:
+        last_poll_time = datetime.now(timezone.utc) - timedelta(seconds=freq_sec)
+        filters = Filters.create(
+            "and",  # need 'and' instead of 'or', even though this is only one filter
+            [
+                Filters.item(
+                    Column.updated_at,
+                    Operator.GREATER_OR_EQUAL,
+                    Type.Datetime,
+                    Filters.value(last_poll_time),
+                )
+            ],
+        )
+        tasks = ls.tasks.list(
+            project=project_id,
+            query=json.dumps({"filters": filters}),
+            fields="all",
+        )
+        yield from tasks
+        time.sleep(freq_seconds)
+
+
+def poll_for_completed_tasks_old(project: Project, freq_seconds: int) -> list:
+    while True:
+        print("polling")
+        last_poll_time = datetime.now(timezone.utc) - timedelta(seconds=freq_seconds)
+        filters = Filters.create(
+            "and",  # need 'and' instead of 'or', even though this is only one filter
+            [
+                Filters.item(
+                    Column.updated_at,
+                    Operator.GREATER_OR_EQUAL,
+                    Type.Datetime,
+                    Filters.value(last_poll_time),
+                )
+            ],
+        )
+        tasks = project.get_tasks(filters=filters)
+        yield from tasks
+        time.sleep(freq_seconds)
+
+
+def perimeter_and_area(source_img_path, annot):
+    width = annot["result"][0]["original_width"]
+    height = annot["result"][0]["original_height"]
+    points = annot["result"][0]["value"]["points"]
+
+    # convert relative coordinates to pixel coordinates
+    points_pxl = [(x / 100 * width, y / 100 * height) for x, y in points]
+
+    with rasterio.open(source_img_path) as src:
+        # convert pixel coordinates to geographic coordinates
+        points_geo = [src.transform * (x, y) for x, y in points_pxl]
+
+    # use Shapely to create a polygon
+    poly = Polygon(points_geo)
+
+    # assume the image CRS is in meters
+    perimeter_m = poly.length
+    area_m = poly.area
+
+    perimeter_km = perimeter_m / 1e3
+    area_km = area_m / 1e6
+
+    return perimeter_km, area_km
+
+
+if __name__ == "__main__":
+    url = "http://localhost:8080"
```

Comment:
is this script supposed to be used in the custom project setup? better to make reproducible setup and instructions for custom input parameters

Reply:
No, this example is without custom scripting. Instructions for setup are in the readme